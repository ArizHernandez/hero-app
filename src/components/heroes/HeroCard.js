import React from 'react'
import { Link } from 'react-router-dom'
import { heroImages } from '../../helpers/heroImages'

export const HeroCard = ({ 
  alter_ego,
  characters,
  first_appearance,
  id,
  superhero
}) => {

  return (
    <div className="card text-center">
      <img 
        src={heroImages(id)} 
        className="card-img-top" 
        alt="..." 
      />

      <div className="card-body">
        <h5 className="card-title">{ superhero }</h5>
        <p className="card-text hero-name">
          {
            ( alter_ego !== characters )
            ? characters
            : alter_ego 
          }
        </p> 
        <p className="card-text text-end mb-2">
          <small className="text-muted">{ first_appearance }</small>
        </p>

        <Link
          to={`./hero/${ id }`}
          className="btn btn-dark btn-block"
        >
          Mas...
        </Link>
      </div>
    </div>
  )
}
