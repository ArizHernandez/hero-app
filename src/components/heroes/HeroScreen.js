import React, { useMemo } from 'react'
import { Redirect, useParams } from 'react-router'

import { getHerosById } from '../../selectors/getHerosById';
import { heroImages } from '../../helpers/heroImages';

export const HeroScreen = ({ history }) => {

  const {heroId} = useParams();
  const hero = useMemo(() => getHerosById({id: heroId}), [heroId]);

  if(!hero || hero.length === 0) {return <Redirect to="/"/>}

  const handleReturn = () => {
    if(history.length <= 2){
      history.push('/')
    } else {
      history.goBack();
    }

  } 

  const [{ superhero, publisher, alter_ego, first_appearance, characters }]  = hero;

  return (
    <div className="container mt-3">
      <button 
        className="btn btn-danger mb-4"
        onClick={ handleReturn }
      >
        Return
      </button>

      <div className="row">
        <div className="shadow-lg p-0 col-12 col-md-4 col-lg-3 animate__animated animate__backInLeft">
          <img 
            className="img-fluid rounded" 
            src={heroImages(heroId)} 
            alt={heroId}
          />
        </div>
        <div className="col text-center">
          <h3 className="display-4 mb-2">{ superhero } <small className="text-muted fs-4"> { publisher }&copy; </small> </h3>
          <p className="mb-4"> { alter_ego } - { first_appearance } </p>
          
          <h3 className="display-6 "> Characters </h3>
          <p> { characters } </p>
        </div>
      </div>
    </div>
  )
}
