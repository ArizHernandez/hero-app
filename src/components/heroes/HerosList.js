import React, { useMemo } from 'react'
import { getHerosByPublisher } from '../../selectors/getHerosByPublisher'
import { HeroCard } from './HeroCard'

export const HerosList = ({ publisher }) => {

  const heroes = useMemo(() => getHerosByPublisher({ publisher }), [ publisher ]);

  return (
    <div className="card-group">
      <div className="row row-cols-1 row-cols-md-3 row-cols-lg-5 g-4">
        { 
          heroes.map( hero => {
            return (
            <div 
              className=" animate__animated animate__fadeIn"
              key={ hero.id }
            >
              <HeroCard { ...hero }/>
            </div>
            )}
          ) 
        }
      </div>
    </div>
  )
}
