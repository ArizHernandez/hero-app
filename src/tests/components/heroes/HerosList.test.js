import { shallow } from "enzyme"
import { HerosList } from "../../../components/heroes/HerosList"

describe('Pruebas en <HerosList />', () => {
  
  const wrapper = shallow(<HerosList publisher={'Marvel Comics'} />)

  test('Debe de cargar correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
})
