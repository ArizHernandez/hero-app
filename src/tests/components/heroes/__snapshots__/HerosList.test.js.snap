// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HerosList /> Debe de cargar correctamente 1`] = `
<div
  className="card-group"
>
  <div
    className="row row-cols-1 row-cols-md-3 row-cols-lg-5 g-4"
  >
    <div
      className=" animate__animated animate__fadeIn"
      key="marvel-spider"
    >
      <HeroCard
        alter_ego="Peter Parker"
        characters="Peter Parker"
        first_appearance="Amazing Fantasy #15"
        id="marvel-spider"
        publisher="Marvel Comics"
        superhero="Spider Man"
      />
    </div>
    <div
      className=" animate__animated animate__fadeIn"
      key="marvel-captain"
    >
      <HeroCard
        alter_ego="Steve Rogers"
        characters="Steve Rogers"
        first_appearance="Captain America Comics #1"
        id="marvel-captain"
        publisher="Marvel Comics"
        superhero="Captain America"
      />
    </div>
    <div
      className=" animate__animated animate__fadeIn"
      key="marvel-iron"
    >
      <HeroCard
        alter_ego="Tony Stark"
        characters="Tony Stark"
        first_appearance="Tales of Suspense #39"
        id="marvel-iron"
        publisher="Marvel Comics"
        superhero="Iron Man"
      />
    </div>
    <div
      className=" animate__animated animate__fadeIn"
      key="marvel-thor"
    >
      <HeroCard
        alter_ego="Thor Odinson"
        characters="Thor Odinson"
        first_appearance="Journey into Myster #83"
        id="marvel-thor"
        publisher="Marvel Comics"
        superhero="Thor"
      />
    </div>
    <div
      className=" animate__animated animate__fadeIn"
      key="marvel-hulk"
    >
      <HeroCard
        alter_ego="Bruce Banner"
        characters="Bruce Banner"
        first_appearance="The Incredible Hulk #1"
        id="marvel-hulk"
        publisher="Marvel Comics"
        superhero="Hulk"
      />
    </div>
    <div
      className=" animate__animated animate__fadeIn"
      key="marvel-wolverine"
    >
      <HeroCard
        alter_ego="James Howlett"
        characters="James Howlett"
        first_appearance="The Incredible Hulk #180"
        id="marvel-wolverine"
        publisher="Marvel Comics"
        superhero="Wolverine"
      />
    </div>
    <div
      className=" animate__animated animate__fadeIn"
      key="marvel-daredevil"
    >
      <HeroCard
        alter_ego="Matthew Michael Murdock"
        characters="Matthew Michael Murdock"
        first_appearance="Daredevil #1"
        id="marvel-daredevil"
        publisher="Marvel Comics"
        superhero="Daredevil"
      />
    </div>
    <div
      className=" animate__animated animate__fadeIn"
      key="marvel-hawkeye"
    >
      <HeroCard
        alter_ego="Clinton Francis Barton"
        characters="Clinton Francis Barton"
        first_appearance="Tales of Suspense #57"
        id="marvel-hawkeye"
        publisher="Marvel Comics"
        superhero="Hawkeye"
      />
    </div>
    <div
      className=" animate__animated animate__fadeIn"
      key="marvel-cyclops"
    >
      <HeroCard
        alter_ego="Scott Summers"
        characters="Scott Summers"
        first_appearance="X-Men #1"
        id="marvel-cyclops"
        publisher="Marvel Comics"
        superhero="Cyclops"
      />
    </div>
    <div
      className=" animate__animated animate__fadeIn"
      key="marvel-silver"
    >
      <HeroCard
        alter_ego="Norrin Radd"
        characters="Norrin Radd"
        first_appearance="The Fantastic Four #48"
        id="marvel-silver"
        publisher="Marvel Comics"
        superhero="Silver Surfer"
      />
    </div>
  </div>
</div>
`;
